libs/application/templates/parental-leave/src/fields/Rights/TransferRights.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { FC, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+
+import { RadioFormField } from '@island.is/application/ui-fields'
+import {
+  FieldBaseProps,
+  FieldComponents,
+  CustomField,
+  FieldTypes,
+} from '@island.is/application/core'
+import { useLocale } from '@island.is/localization'
+
+import { parentalLeaveFormMessages, errorMessages } from '../../lib/messages'
+import { getApplicationAnswers } from '../../lib/parentalLeaveUtils'
+import { maxDaysToGiveOrReceive } from '../../config'
+import { YES, NO, TransferRightsOption } from '../../constants'
+import { YesOrNo } from '../../types'
+
+const getDefaultValue = (
+  isRequestingRights: YesOrNo,
+  isGivingRights: YesOrNo,
+) => {
+  if (isRequestingRights === YES && isGivingRights === YES) {
+    return undefined
+  }
+
+  if (isRequestingRights === YES) {
+    return TransferRightsOption.REQUEST
+  }
+
+  if (isGivingRights === YES) {
+    return TransferRightsOption.GIVE
+  }
+
+  if (isRequestingRights === NO && isGivingRights === NO) {
+    return TransferRightsOption.NONE
+  }
+
+  return undefined
+}
+
+const clampDayValue = (value: number, min: number, max: number) => {
+  return Math.min(Math.max(min, value), max)
+}
+
+interface HiddenValues {
+  isRequestingRights: YesOrNo
+  requestDays: number
+  isGivingRights: YesOrNo
+  giveDays: number
+}
+
+const calculateHiddenValues = (
+  selectedOption: TransferRightsOption | undefined,
+  requestDays: number,
+  giveDays: number,
+): HiddenValues | undefined => {
+  if (selectedOption === TransferRightsOption.REQUEST) {
+    return {
+      isRequestingRights: YES,
+      requestDays: clampDayValue(requestDays, 1, maxDaysToGiveOrReceive),
+      isGivingRights: NO,
+      giveDays: 0,
+    }
+  } else if (selectedOption === TransferRightsOption.GIVE) {
+    return {
+      isGivingRights: YES,
+      giveDays: clampDayValue(giveDays, 1, maxDaysToGiveOrReceive),
+      isRequestingRights: NO,
+      requestDays: 0,
+    }
+  } else if (selectedOption === TransferRightsOption.NONE) {
```

Comment:
Could this be the default return value instead of undefined?

Reply:
That way we can get rid of conditions below.

Reply:
Yes, that will probably work 👍